src/test/acceptance/test_authenticated_gui.py

```diff
@@ -1,68 +1,135 @@
+            # t_superuser,       superuser,     test,  k2GKnNaA5UlENStVI4AEJKQ7BP9ZqO+21Cx746BjJDo=
+            'user-database': ''.join(['sqlite:///', get_test_data_dir(), '/user_test.db']),
+        },
+    }
+)
+class TestAcceptanceAuthentication:
     UNIQUE_LOGIN_STRING = b'<h3 class="mx-3 mt-4">Login</h3>'
     PERMISSION_DENIED_STRING = b'You do not have permission to view this resource.'
 
-    def test_redirection(self):
-        response = self.test_client.get('/', follow_redirects=False)
-        self.assertIn(b'Redirecting', response.data, 'no redirection taking place')
-
-    def test_show_login_page(self):
-        response = self.test_client.get('/', follow_redirects=True)
-        self.assertIn(self.UNIQUE_LOGIN_STRING, response.data, 'no authorization required')
-
-    def test_api_key_auth(self):
-        response = self.test_client.get('/', headers={'Authorization': self.guest.key}, follow_redirects=True)
-        self.assertNotIn(self.UNIQUE_LOGIN_STRING, response.data, 'authorization not working')
-
-    def test_role_based_access(self):
-        self._start_backend()
-        try:
-            response = self.test_client.get('/upload', headers={'Authorization': self.guest.key}, follow_redirects=True)
-            self.assertIn(self.PERMISSION_DENIED_STRING, response.data, 'upload should not be accessible for guest')
-
-            response = self.test_client.get(
-                '/upload', headers={'Authorization': self.guest_analyst.key}, follow_redirects=True
-            )
-            self.assertIn(
-                self.PERMISSION_DENIED_STRING, response.data, 'upload should not be accessible for guest_analyst'
-            )
-
-            response = self.test_client.get(
-                '/upload', headers={'Authorization': self.superuser.key}, follow_redirects=True
-            )
-            self.assertNotIn(self.PERMISSION_DENIED_STRING, response.data, 'upload should be accessible for superusers')
-        finally:
-            self._stop_backend()
-
-    def test_about_doesnt_need_authentication(self):
-        response = self.test_client.get('/about', follow_redirects=True)
-        self.assertNotIn(self.UNIQUE_LOGIN_STRING, response.data, 'authorization required')
+    def test_redirection(self, test_client):
+        response = test_client.get('/', follow_redirects=False)
+        assert b'Redirecting' in response.data, 'no redirection taking place'
+
+    def test_show_login_page(self, test_client):
+        response = test_client.get('/', follow_redirects=True)
+        assert self.UNIQUE_LOGIN_STRING in response.data, 'no authorization required'
+
+    def test_api_key_auth(self, test_client):
+        response = test_client.get('/', headers={'Authorization': guest.key}, follow_redirects=True)
+        assert self.UNIQUE_LOGIN_STRING not in response.data, 'authorization not working'
+
+    @pytest.mark.usefixtures('start_backend')
+    def test_role_based_access(self, frontend, test_client):
```

Comment:
The fixture always returns `None`. So having it as an argument is not needed.
Also having it as an argument causes pylint warnings because it is unused. Using `@pytest.mark.usefixtures` is more clear imo.